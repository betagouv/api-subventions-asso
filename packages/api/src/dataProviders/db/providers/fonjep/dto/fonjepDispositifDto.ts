export default interface FonjepDispositifDto {
    ID: number;
    Libelle: string;
    FinanceurCode: string;
}
