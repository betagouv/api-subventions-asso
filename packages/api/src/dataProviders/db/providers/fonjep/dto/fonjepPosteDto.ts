export default interface FonjepPosteDto {
    [key: string]: any;
    // dates are excel dates (i.e. days since 1900-01-01)
    Code: string | null;
    DispositifId: number | null;
    PstStatutPosteLibelle: string | null;
    PstRaisonStatutLibelle: string | null;
    FinanceurPrincipalCode: string | null;
    FinanceurAttributeurCode: string | null;
    AssociationBeneficiaireCode: string | null;
    AssociationImplantationCode: string | null;
    Annee: number | null;
    MontantSubvention: number | null;
    DateFinTriennalite: number | null;
    PstTypePosteCode: string | null;
    PleinTemps: string | null;
    DoublementUniteCompte: string | null;
}
