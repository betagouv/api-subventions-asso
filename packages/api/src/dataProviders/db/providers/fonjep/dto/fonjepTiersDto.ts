export default interface FonjepTiersDto {
    [key: string]: any;

    Code: string | null;
    RaisonSociale: string | null;
    EstAssociation: string | null;
    EstCoFinanceurPostes: string | null;
    EstFinanceurPostes: string | null;
    SiretOuRidet: string | null;
    CodePostal: string | null;
    Ville: string | null;
    ContactEmail: string | null;
}
