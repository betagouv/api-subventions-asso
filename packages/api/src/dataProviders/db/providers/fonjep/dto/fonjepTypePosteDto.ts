export default interface FonjepTypePosteDto {
    Code: string | null;
    Libelle: string | null;
}
