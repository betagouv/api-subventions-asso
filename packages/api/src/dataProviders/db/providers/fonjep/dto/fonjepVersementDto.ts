// null represents an empty cell in the file
export default interface FonjepVersementDtoWithExcelDate {
    PosteCode: string;
    PeriodeDebut: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    PeriodeFin: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    DateVersement: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    MontantAPayer: number | null;
    MontantPaye: number | null;
}

export interface FonjepVersementDto
    extends Omit<FonjepVersementDtoWithExcelDate, "DateVersement" | "PeriodeDebut" | "PeriodeFin"> {
    PeriodeDebut: Date | null;
    PeriodeFin: Date | null;
    DateVersement: Date | null;
}
