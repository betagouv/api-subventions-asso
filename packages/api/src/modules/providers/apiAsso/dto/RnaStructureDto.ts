export interface RnaStructureDto {
    identite: {
        nom: string;
        sigle: string;
        id_rna: string;
        date_pub_jo: string;
        date_modif_rna: string;
        active: boolean;
        nature: string;
    };
    activites: {
        objet: string;
        lib_objet_social1: string;
        annee_activite_principale: number;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            cp: number;
            bp?: string;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            voie: string;
            commune: string;
            cp: number;
            pays: string;
        };
    };
}
