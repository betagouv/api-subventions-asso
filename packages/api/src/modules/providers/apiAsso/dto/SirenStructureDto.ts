export interface SirenStructureEtablissementDto {
    actif: boolean;
    adresse: {
        cplt_1?: string;
        num_voie: number;
        voie: string;
        commune: string;
        code_insee: number;
        cp: number;
        type_voie: string;
    };
    id_siret: number;
    est_siege: boolean;
    date_actif: string;
    id_activite_principale: string;
    annee_activite_principale: number;
    effectif_salarie_cent: number;
    id_tranche_effectif?: string;
    annee_effectif_salarie_cent: number;
}

export interface SirenStructureDto {
    identite: {
        nom: string;
        nom_sirene: string;
        id_siren: number;
        id_siret_siege: number;
        id_forme_juridique: number;
        date_creation_sirene: string;
        date_modif_siren: string;
        active: boolean;
        active_sirene: boolean;
        regime: string;
    };
    activites: {
        id_activite_principale: string;
        annee_activite_principale: number;
        id_tranche_effectif: string;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
        appartenance_ess: string;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
    };
    nbEtabsActifs: number;
    etablissement?: SirenStructureEtablissementDto[];
}
