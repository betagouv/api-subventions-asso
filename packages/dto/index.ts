export * from "./etablissements/index";
export * from "./versements/index";
export * from "./user/index";
export * from "./auth/LoginDtoResponse";
export * from "./auth/ResetPasswordDtoResponse";
export * from "./auth/SignupDtoResponse";
export * from "./associations/Association";
export * from "./search/DemandeSubvention";
export * from "./associations/GetAssociationResponseDto";
export * from "./associations/GetEtablissementsResponseDto";
export * from "./associations/GetEtablissementResponseDto";
export * from "./associations/GetSubventionResponseDto";
export * from "./associations/GetSubventionsResponseDto";
export * from "./associations/GetVersementsResponseDto";
export * from "./associations/GetDocumentsResponseDto";
export * from "./shared/ProviderValue";
export * from "./shared/Rna";
export * from "./shared/Siren";
export * from "./shared/Siret";
export * from "./shared/Adresse";
export * from "./shared/Personne";
export * from "./shared/InformationBancaire";
export * from "./user/UserDto";
export * from "./stats/RequestsDtoResponse";
export * from "./stats/StatsRequestsMedianDto";
export * from "./stats/MonthlyAvgRequestDtoResponse";
export * from "./rna-siren/RnaSirenResponseDts";
export * from "./consumer/ConsumerTokenDto";
