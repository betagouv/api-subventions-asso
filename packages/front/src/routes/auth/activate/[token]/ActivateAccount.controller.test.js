import { nanoid } from "nanoid";
import { AgentTypeEnum } from "dto";
import ActivateAccountController from "./ActivateAccount.controller";
import authService from "$lib/resources/auth/auth.service";
import { goToUrl } from "$lib/services/router.service.js";

vi.mock("$lib/resources/auth/auth.service", () => ({
    default: {
        activate: vi.fn(async () => ({})),
    },
}));
vi.mock("$lib/services/router.service");

describe("ActivateAccountController", () => {
    let controller;

    const FAKE_TOKEN = nanoid();

    afterEach(() => (controller = null));

    describe("constructor", () => {
        beforeEach(() => (controller = new ActivateAccountController(FAKE_TOKEN)));
        it("should define form steps", () => {
            const actual = controller.steps;
            expect(actual).toMatchSnapshot();
        });
    });

    describe("onSubmit", () => {
        const USER = {
            agentType: AgentTypeEnum.OPERATOR,
            something: "something",
        };

        it("should call authService.activate() with given values", async () => {
            controller = new ActivateAccountController(FAKE_TOKEN);
            await controller.onSubmit(USER);
            expect(authService.activate).toHaveBeenCalledWith(FAKE_TOKEN, USER);
        });

        it("should redirect to successful login", async () => {
            controller = new ActivateAccountController(FAKE_TOKEN);
            await controller.onSubmit(USER);
            expect(goToUrl).toHaveBeenCalledWith("/auth/login?success=ACCOUNT_ACTIVATED", false, true);
        });
    });
});
