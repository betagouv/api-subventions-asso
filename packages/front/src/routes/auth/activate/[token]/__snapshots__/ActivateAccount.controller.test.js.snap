// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActivateAccountController > constructor > should define form steps 1`] = `
[
  {
    "alert": [Function],
    "component": [Function],
    "name": "DÃ©finir un mot de passe",
  },
  {
    "alert": [Function],
    "component": [Function],
    "name": "Informations sur votre profil",
  },
]
`;
